### PR TITLE
Add upgrade tests for HA

### PR DESCRIPTION
**Pull request by ldevulder**
This commit adds code to be able to upgrade a HA cluster with all HA stack/resources configured, and test that all is still running well after the upgrade.

Verification run:
* http://1b147.qa.suse.de/tests/2508
* http://1b147.qa.suse.de/tests/2511
* http://1b147.qa.suse.de/tests/2537
* http://1b147.qa.suse.de/tests/2539

### CODE REVIEW

File: lib/hacluster.pm [hunk 1]
```diff
@@ -228,8 +229,31 @@ sub ha_export_logs {
...
+        # This can takes time, so a loop is a good idea here
+        while ($count > 0) {
+            # Check for cluster/resources status and exit loop when needed
+            $ret = script_run "$cmd";
+            last if (defined $ret and $ret == 0);
+
+            # Otherwise wait a while
+            sleep($count--);
+        }
+        $count = 10;    # Reset counter
```
[the remark this answers is not in the document]
Comment (ldevulder) [reply lib/hacluster.pm]: You're right!

File: lib/hacluster.pm [hunk 2]
```diff
@@ -228,8 +229,31 @@ sub ha_export_logs {
...
+    # Execute each comnmand to validate that the cluster is running
+    foreach my $cmd (@cmds) {
+        # This can takes time, so a loop is a good idea here
+        while ($count > 0) {
+            # Check for cluster/resources status and exit loop when needed
+            $ret = script_run "$cmd";
+            last if (defined $ret and $ret == 0);
+
+            # Otherwise wait a while
+            sleep($count--);
```
[the remark this answers is not in the document]
Comment (ldevulder) [reply lib/hacluster.pm]: As cluster can  takes time to complete its initialization I think that wait for smaller period first and then increase can lead to wait for at least  (1+2+3+4+5) seconds so worse than 10s. Static sleep may be better yes, I will try.

File: lib/hacluster.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (ldevulder) [reply lib/hacluster.pm]: Yes, it's better :-)

File: lib/hacluster.pm (hunk 2 above)
[the remark this answers is not in the document]
Comment (ldevulder) [reply lib/hacluster.pm]: Should be better now with this new version.

File: lib/hacluster.pm [hunk 3]
```diff
@@ -228,8 +229,36 @@ sub ha_export_logs {
...
+    my $ret       = undef;
+
+    # Execute each comnmand to validate that the cluster is running
+    # This can takes time, so a loop is a good idea here
+    foreach my $cmd (@cmds) {
+        # Check for cluster/resources status and exit loop when needed
+        while ($ret = script_run "$cmd") {
+            # Otherwise wait a while if timeout is not reached
+            my $timerun = time - $starttime;
+            if ($timerun < $timeout) {
```
Comment (rwx788) [inline lib/hacluster.pm]: As of now I see that timeout is for all commands in the array, could you please put a comment somewhere. I like this approach. As below message in die says can be interpreted that single command failed in 120 seconds. So maybe you wanted to set `$starttime = time` before each iteration.

Comment (ldevulder) [reply lib/hacluster.pm]: Fixed!

Comment (rwx788) [reply lib/hacluster.pm]: Thanks!

File: lib/hacluster.pm [hunk 4]
```diff
@@ -228,8 +229,39 @@ sub ha_export_logs {
...
+    assert_script_run 'crm_verify -LV';
+}
+
+# Wait for resources to be started
+sub wait_until_resources_started {
+    my @cmds    = ('crm cluster wait_for_startup', "$crm_mon_cmd | grep -i 'no inactive resources'");
+    my $timeout = 120;
+    my $ret     = undef;
+
+    # Execute each comnmand to validate that the cluster is running
```
Comment (ldevulder) [inline lib/hacluster.pm]: Ouch ! I just saw a mistake, 'comnmand' instead of 'command'...

As the PR is merged I will try to correct this in a next PR :)

Comment (rwx788) [reply lib/hacluster.pm]: True, I guess we need built -in spell checker. @okurz is really good in catching those =)

**Pull request merged**